### PR TITLE
chore: add issue and pr template 

### PR DESCRIPTION
**Pull request by mlnwns**
## 🚀 Pull Request Description
I've created comprehensive issue and pull request templates for the function-al library to improve our contribution process. These templates are designed to provide clear guidance for contributors and maintainers, ensuring that all submissions include essential information.

The key objectives of these templates are to:
- Standardize the format of issue and pull request submissions
- Provide clear instructions for contributors
- Capture important details about proposed changes
- Align with our existing Contributing Guide

I've developed templates for different types of contributions, including:
- Feature requests
- Bug reports
- Test case additions
- Documentation updates
- Chore and infrastructure improvements

Each template includes sections that prompt contributors to provide:
- A clear description of their proposed changes
- Context and motivation
- Testing details
- Environment information
- A checklist to ensure all contribution guidelines are followed

These templates will help streamline our contribution process, making it easier for both contributors and maintainers to understand and review proposed changes.

### CODE REVIEW

File: .github/ISSUE_TEMPlATE/bug_report.yml [hunk 1]
```diff
@@ -0,0 +1,64 @@
+name: "\U0001F41E Bug Report"
+description: Report a bug in function-al
+labels: ["bug"]
+body:
+  - type: markdown
```
Comment (hwanheejung) [inline .github/ISSUE_TEMPlATE/bug_report.yml]: I think this section could be removed, and instead, you might consider adding the following checklist at the bottom of the form.

```yml
- type: checkboxes
  id: checklist
  attributes:
    label: Checklist
    description: Please confirm before submitting
    options:
      - label: I have checked that this issue does not already exist.
        required: true
      - label: I have read the [Contributing Guide](https://github.com/hwanheejung/function-al/blob/main/CONTRIBUTING.md).
        required: true
      - label: I am using the latest version of function-al.
        required: false
```

Comment (hwanheejung) [reply .github/ISSUE_TEMPlATE/bug_report.yml]: It might be a good idea to apply this to all issue templates for consistency. 👍

File: .github/ISSUE_TEMPlATE/test_case.yml [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+      label: Test Case Examples
+      description: Provide example test code or pseudocode if possible
+      placeholder: |
+        ```typescript
+        test('function name - specific scenario', () => {
+          // Test case details
+        })
+        ```
+
+  - type: checkboxes
```
Comment (hwanheejung) [inline .github/ISSUE_TEMPlATE/test_case.yml:50]: I think this part can be removed :) cuz we only do unit tests

File: .github/ISSUE_TEMPlATE/test_case.yml [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+      label: Test Types
+      description: Select the types of tests you're proposing
+      options:
+        - label: Unit Tests
+        - label: Edge Case Tests
+        - label: Input Validation Tests
+        - label: Type Coverage Tests
+        - label: Performance Tests
+
+  - type: textarea
```
Comment (hwanheejung) [inline .github/ISSUE_TEMPlATE/test_case.yml]: this part can be removed also

**Review by hwanheejung — CHANGES_REQUESTED** [review]
Thank you so much for this meaningful PR and even more for becoming the first contributor to `function-al`! 👍 💯 

Really appreciate you jumping in :) I left a few small comments. nothing major, mostly just for clarity and consistency.

Once those are updated, I think we’re good to go and can get this merged in 😊

Let me know if you have any questions at all!

**Review by hwanheejung — APPROVED** [review]
LGTM 👍
